sqlmesh/core/selector.py

```diff
@@ -91,6 +91,12 @@ def select_models(
         )
 
         dag: DAG[str] = DAG()
+        subdag = set()
+
+        for fqn in all_selected_models:
+            subdag.add(fqn)
+            subdag.update(self._dag.downstream(fqn))
```

Comment:
This may get rather expensive given the selector expression itself may include downstream through the trailing +.